File: qualifier-week-4/graded-assignments/GrPA8 - Graded.c
#include <stdio.h>

int main() {
    float price, total = 0.0;
    char choice;

    do {
        scanf("%f", &price);

        total += price;

        scanf(" %c", &choice);
    } while (choice == 'y' || choice == 'Y');

    printf("%.2f\n", total);

    return 0;
}

File: qualifier-week-4/graded-assignments/GrPA8 - Graded.rs
use std::io;

fn main() {
    let mut total: f64 = 0.0;
    let mut input = String::new();

    loop {
        io::stdin().read_line(&mut input).expect("Failed to Read Input: Failure");
        let price: f64 = input.trim().parse().expect("Invalid input");

        total += price;

        input.clear();

        io::stdin().read_line(&mut input).expect("Failed to read line: Failure");
        let choice = input.trim();

        input.clear();

        if choice != "y" && choice != "Y" {
            break;
        }
    }

    println!("{:.2}", total);
}
